services/server/src/handlers/report.ts

```diff
@@ -0,0 +1,140 @@
+import { reports } from "@litespace/models";
+import { forbidden, largeFileSize, notfound } from "@/lib/error";
+import { Request, Response } from "express";
+import { NextFunction } from "express";
+import safeRequest from "express-async-handler";
+import {
+  boolean,
+  dateFilter,
+  number,
+  pageNumber,
+  pageSize,
+  string,
+  id,
+} from "@/validation/utils";
+import { isAdmin, isUser } from "@litespace/utils/user";
+import zod from "zod";
+import { paginationDefaults } from "@/constants";
+import { IReport } from "@litespace/types";
+import { exceedsSizeLimit, getRequestFile, upload } from "@/lib/assets";
+import { v4 as uuid } from "uuid";
+
+const createReportPayload = zod.object({
+  title: string,
+  description: string,
+});
+
+const updateReportPayload = zod.object({ id, resolved: boolean });
+
+const findReportsPayload = zod.object({
+  ids: zod.array(number).describe("filter by reports ids").optional(),
+  users: zod.array(number).describe("filter by users ids").optional(),
+  title: string.describe("filter by title").optional(),
+  description: string.describe("filter by description").optional(),
+  screenshot: boolean
+    .describe("filter by screenshot (uploaded or not)")
+    .optional(),
+  log: boolean.describe("filter by log files (uploaded or not)").optional(),
+  resolved: boolean.describe("filter by resolved value").optional(),
+  createdAt: dateFilter.describe("filter by creation time").optional(),
+  updatedAt: dateFilter.describe("filter by upate time").optional(),
+  page: pageNumber.optional().default(paginationDefaults.page),
+  size: pageSize.optional().default(paginationDefaults.size),
+});
+
+async function create(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  const allowed = isUser(user);
+  if (!allowed) return next(forbidden());
+
+  const { title, description }: IReport.CreateApiPayload =
+    createReportPayload.parse(req.body);
+
+  const key = uuid();
+
+  // upload the screenshot if exists
+  const screenshot = getRequestFile(
+    req.files,
+    IReport.AssetFileName.Screenshot
+  );
+  if (screenshot && exceedsSizeLimit(screenshot.size) !== false)
+    return next(largeFileSize());
+  const screenshotKey = screenshot
+    ? await upload({
+        data: screenshot.buffer,
+        type: screenshot.mimetype,
+        prefix: "reports/screenshots/",
+        key,
+      })
+    : undefined;
+
+  // upload the log file if exists
+  const log = getRequestFile(req.files, IReport.AssetFileName.Log);
+  if (log && exceedsSizeLimit(log.size) !== false) return next(largeFileSize());
+  const logKey = log
```

Comment:
empty lines for redablity. 
```suggestion
  // upload the log file if exists
  const log = getRequestFile(req.files, IReport.AssetFileName.Log);
  
  if (log && exceedsSizeLimit(log.size) !== false) return next(largeFileSize());
  
  const logKey = log
```